sklearn/preprocessing/_data.py

```diff
@@ -967,7 +970,7 @@ def partial_fit(self, X, y=None, sample_weight=None):
         # for backward-compatibility, reduce n_samples_seen_ to an integer
         # if the number of samples is the same for each feature (i.e. no
         # missing values)
-        if np.ptp(self.n_samples_seen_) == 0:
+        if np.ptp(self.n_samples_seen_) == 0:  # TODO add ptp to array api utils
```

Comment:
Probably the fallback will be something like `xp.max(self.n_samples_seen_) - xp.min(self.n_samples_seen_)`.

Reply:
That's certainly more elegant. :)

I guess the reason why ptp was used in the first place was because we only have to iterate over self.n_samples_seen_ once. Doing it twice could make the code slower. I just don't know whether it is significant.

Reply:
I think it's fine because this is an iteration over the features/columns that are typically small to medium size in an ML setup.

I am not even sure that `np.ptp` would do a single pass.